src/proto/diff.proto

```diff
@@ -66,6 +66,15 @@ message ChangedFile {
 		CS4 = 204;
 		/** @exclude */
 		CS5 = 205;
+		
+		/** The file represents a JavaScript source file that had a parse error */
+		SOURCE_JS_ERROR = 300;
+		/** The file represents a JavaScript source file that parsed without error as JLS2 */
+		SOURCE_JS_JLS2 = 301;
```

Comment:
I dont think JavaScript is called 'JLS'.  It is defined by ECMAScript versions.  So the enums here should probably be named accordingly (SOURCE_JS_ECMA1/2/3/4/5/6/7 etc).  Also, which versions do we support?  Perhaps only include constants for those supported.